examples/vello_editor/src/text.rs

```diff
@@ -126,11 +168,54 @@ impl Editor {
         // TODO: support clipboard on Android
     }
 
-    pub fn handle_event(&mut self, event: &WindowEvent) {
+    /// Suggest an area for IME candidate box placement based on the current IME state.
+    fn set_ime_cursor_area(&self, window: &Window) {
+        if let Some(ref text_range) = self.preedit_range {
+            // Find the smallest rectangle that contains the entire preedit text.
+            // Send that rectangle to the platform to suggest placement for the IME
+            // candidate box.
```

Comment:
Hmm, I'm not sure that this is quite the right logic. For example, consider the Windows emoji picker; at least ideally it uses the cursor area, but doesn't add a pre-edit text.
I think that this whole statement needs to have an else branch, which uses something based on the selection.
My proposal would be to use one of:
1) The entire selection area;
2) The area of the `focus`ed cluster

Reply:
I think you're right. Perhaps the area should always be based on the selection: either there is no preedit, and the selection is the only thing that makes sense, or there is a preedit, and the IME gave us a selection within it (`Ime::Preedit(_, cursor)`). Although, if the preedit wraps lines and the preedit cursor is a partial selection, the candidate box may then overlap.

I'm also a bit confused by the name [`Window::set_ime_cursor_area`](https://docs.rs/winit/latest/winit/window/struct.Window.html#method.set_ime_cursor_area), as that seems to suggest the area of the IME cursor (as given by `Ime::Preedit(_, cursor)`) is to be given, whereas the documentation states: "an example of such area could be a input field in the UI or line in the editor."

Reply:
Yeah; this API is underdocumented from the winit side. I don't have good answers here.

I think the IME box should never cover the pre-edit text, so if there is a pre-edit, I think the current answer is fine.

Reply:
The area is now calculated as the minimum rect containing the preedit text, if present. If not, it depends on whether the selection is collapsed. A collapsed selection sends the caret area, a spanning selection is the minimum rect containing the selection.